client/me/purchases/billing-history/field-definitions.tsx

```diff
@@ -22,7 +22,8 @@ function renderServiceNameDescription(
 	const plan = capitalPDangit( transaction.variation );
 	const termLabel = getTransactionTermLabel( transaction, translate );
 	const shouldShowDomain =
-		transaction.domain && ! transaction.domain.startsWith( 'siteless.marketplace.wp.com' );
+		transaction.domain &&
+		! /^siteless.(marketplace.wp|agencies.automattic|a4a).com/.test( transaction.domain );
```

Comment:
You may want to escape the `.` characters so they are treated as periods and not as wildcards.

```suggestion
		! /^siteless\.(marketplace\.wp|agencies\.automattic|a4a)\.com/.test( transaction.domain );
```

Not as important but this condition might be worth putting into its own variable for readability.

Reply:
Great catch - thank you! I just made the updates.